File: 2025/03/test_main.lua
local lib = require("lib")

---@param expected integer
---@param actual integer
local function assert(expected, actual)
    if expected == actual then
        print("Assertion Passed.")
    else
        print(string.format("Assertion Failed; expected: %d, actual: %d", expected, actual))
    end
end

function test_main()
    local raw = "987654321111111\n811111111111119\n234234234234278\n818181911112111\n"

    ---@type string[]
    local data = {}

    for line in raw:gmatch("[^\n]+") do
        table.insert(data, line)
    end

    local first_expected = 357
    local first_result = lib.first(data)
    assert(first_expected, first_result)

    local second_expected = 3121910778619
    local second_result = lib.second(data)
    assert(second_expected, second_result)
end

test_main()

File: 2025/02/test_main.py
from lib import first, second


def test_main():
    raw = "11-22,95-115,998-1012,1188511880-1188511890,222220-222224,1698522-1698528,446443-446449,38593856-38593862,565653-565659,824824821-824824827,2121212118-2121212124"
    data = raw.split(",")

    first_expected = 1227775554
    first_result = first(data)
    assert first_expected == first_result

    second_expected = 4174379265
    second_result = second(data)
    assert second_expected == second_result
